doc/development/how-to/react-hook-form.md

```diff
@@ -0,0 +1,82 @@
+# React-hook-form
+
+## Les composants réutilisables
+
+### Les controllers
+
+On les trouve dans `src/shared/forms/controllers`.
+
+Ils servent à être utiliser dans les formulaires utilisant react-hook-form et remplacent les inputs classiques.
+
+### Les schemas
+
+On les trouve dans `src/shared/forms/schemas`.
+
+Ils regroupent des règles de validation classique qu'on retrouve à plusieurs endroits de l'application comme les mots de passe ou adresse email et peuvent être utiliser dans d'autres schemas de cette manière-ci :
+
+```ts
+export const setEmailSchema = object()
+  .shape({
+    email: emailSchema.required('L’email est obligatoire'),
+  })
+  .required()
+```
+
+## Tester un formulaire
+
+### Tester le schema yup
+
+Pour tester un schema yup, il faut passer par la fonction validate du schema, le résultat vaut soit la valeur d'entrée dans le cas ou le schema est valide, soit rejects avec le message d'erreur dans le cas où le schema n'est pas valide.
+
+```tsx
+// Cas valide
+it('must follow all security rules', async () => {
+  const value = 'user@AZERTY123'
+  const result = await passwordSchema.validate(value)
+
+  expect(result).toEqual(value)
+})
+
+// Cas invalide
+it('must have at least 1 special character', async () => {
+  const value = 'userAZERTY123'
+  const result = passwordSchema.validate(value)
+
+  await expect(result).rejects.toEqual(new ValidationError('1 Caractère spécial (!@#$%^&*...)'))
+})
+```
```

Comment:
Je peux comprendre qu'on ai envie de décrire les types de cas qu'on test

Mais je trouverais ça dommage d'avoir des commentaires dans le code qui décrivent ce qu'est censé décrire le `it`

Un copié / collé est vite arrivé, je recommande de ne pas les mettre dans la doc

~~~suggestion
Cas valide

```tsx
it('must follow all security rules', async () => {
  const value = 'user@AZERTY123'
  const result = await passwordSchema.validate(value)

  expect(result).toEqual(value)
})
```

Cas invalide

```tsx
it('must have at least 1 special character', async () => {
  const value = 'userAZERTY123'
  const result = passwordSchema.validate(value)

  await expect(result).rejects.toEqual(new ValidationError('1 Caractère spécial (!@#$%^&*...)'))
})
```
~~~